validphys2/src/validphys/tests/test_pseudodata.py

```diff
@@ -2,111 +2,81 @@
 Test to ensure the validphys.pseudodata.get_pseudodata action
 correctly obtains the appropriate pseudodata for an n3fit fit.
 
-To this end, a 10 replica fit has been uploaded named
-`pseudodata_test_fit` obtained using 100 epochs, theoryID 162 and a
-subset of DIS datasets. When this fit was performed, the `all_exp_infos`
-was pickled and stored in `exp_infos.pickle` which is the benchmark
-we use to ensure the action is working appropriately.
+A fit has been generated called pseudodata_test_fit_n3fit
+which has the pseudodata saved as training and validation splits.
+This is used to benchmark the correctness of the pseudodata
+recreation.
 """
-import pickle
-from importlib.resources import read_binary
-
-import numpy as np
+import pandas as pd
 import pytest
 
 from validphys.api import API
-from validphys.pseudodata import training_validation_pseudodata
-import validphys.tests.regressions
+from validphys.tests.conftest import FIT
 
-from reportengine.checks import CheckError
-from reportengine.compat import yaml
-from reportengine.resourcebuilder import ResourceError
+PSEUDODATA_FIT = "pseudodata_test_fit_n3fit"
 
-EXAMPLE_RUNCARD = """fit: pseudodata_test_fit
-pdf: pseudodata_test_fit
 
-experiments:
-  from_: fit
+def test_read_fit_pseudodata():
+  fit_pseudodata = API.read_fit_pseudodata(fit=PSEUDODATA_FIT)
 
-t0pdfset:
-  from_: datacuts
+  nrep = API.num_fitted_replicas(fit=PSEUDODATA_FIT)
+  assert nrep == len(fit_pseudodata)
 
-datacuts:
-  from_: fit
+  for data, tr_idx, val_idx  in fit_pseudodata:
+    assert set(tr_idx).isdisjoint(set(val_idx))
+    assert set(tr_idx).union(val_idx) == set(data.index)
 
-use_cuts: fromfit
-"""
 
+def test_read_pdf_pseudodata():
+  pdf_pseudodata = API.read_pdf_pseudodata(fit=PSEUDODATA_FIT)
 
-@pytest.fixture(
-    scope="session",
-    params=[1, pytest.param(None, marks=pytest.mark.linux)],
-)
-def setup_dicts(request):
-    n_process_config = dict(NPROC=request.param)
-    exp_infos_bytes = read_binary(validphys.tests.regressions, "test_exp_infos.pickle")
-    ns = yaml.safe_load(EXAMPLE_RUNCARD)
-    # This is what all the fitted replicas saw
-    exp_infos = pickle.loads(exp_infos_bytes)
+  pdf = API.pdf(pdf=PSEUDODATA_FIT)
+  # -1 because we ignore replica 0
+  assert len(pdf) - 1 == len(pdf_pseudodata)
```

Comment:
Please change the indentation to 4 spaces.

Reply:
Good spot, I've now reformatted this entire file using black